source/nanoFramework.Hardware.Esp32.Rmt/TransmitterChannel.cs

```diff
@@ -71,97 +80,130 @@ public int Channel
-            {
-                if (value > MaxChannelNumber)
-                {
-                    throw new ArgumentOutOfRangeException("Channel", $"The maximum channel number is {MaxChannelNumber}");
-                }
+            private set
+            {                
                 _channel = value;
             }
         }
 
-        public byte ClockDivider
+        /// <summary>
+        /// Add new rmt command to the list of commands that will be send
+        /// </summary>
+        /// <param name="cmd">RmtCommand</param>
+        public void AddCommand(RmtCommand cmd)
         {
-            get
-            {
-                return NativeGetClockDivider(Channel);
-            }
-            set
-            {
-                if (value == 0)
-                {
-                    throw new ArgumentOutOfRangeException("ClockDivider", "Must be in the range 1..255");
-                }
-                NativeSetClockDivider(Channel, value);
-            }
+            commands.Add(cmd);
         }
 
         /// <summary>
-        /// True if the channel's memory is in low power mode; false otherwise.
+        /// Access any of the commands from the list that will be send
         /// </summary>
-        public bool IsMemoryInLowPowerMode
+        /// <param name="i"></param>
+        /// <returns></returns>
+        public RmtCommand this[int i]
         {
             get
             {
-                return NativeGetMemoryLowPower(Channel);
+                if (commands.Count < i + 1)
+                    throw new IndexOutOfRangeException();
+
+                var res = commands[i];
+                return (RmtCommand)res;
             }
             set
             {
-                NativeSetMemoryLowPower(Channel, value);
+                if (commands.Count < i + 1)
+                    throw new IndexOutOfRangeException();
+
+                commands[i] = value;
             }
         }
 
-        /// <summary>
-        /// The number of 64 * 32-bit memory blocks held by the channel.
-        /// </summary>
-        /// <remarks>
-        /// The 8 channels share a pool of 512 32-bit memory blocks.
-        /// If a channel has more than one memory blocks, it will occupy
-        /// the memory blocks of subsequent channels. Thus, Channel 0 can
-        /// use at most 8 blocks and channel 7 one.
-        /// </remarks>
-
-        public byte MemoryBlockNumber
+        public byte ClockDivider
         {
             get
             {
-                return NativeGetMemoryBlockNumber(Channel);
+                return NativeGetClockDivider(Channel);
             }
             set
             {
-                if (value > (NumberRmtChannels - Channel))
+                if (value == 0)
                 {
-                    throw new ArgumentOutOfRangeException("MemoryBlockNumber", "");
+                    throw new ArgumentOutOfRangeException("ClockDivider", "Must be in the range 1..255");
```

Comment:
Need to save every possible byte in flash space. 
Move the strings to a comment and add exception entry in the documentation comment with it too.